test/opt/inline_test.cpp

```diff
@@ -0,0 +1,1071 @@
+  };
+  SinglePassRunAndCheck<opt::InlinePass>(JoinAllInsts(before),
+                                                 JoinAllInsts(after),
+                                                 /* skip_nop = */ false);
+}
+
+TEST_F(InlineTest, Nested) {
+  const std::vector<const char*> before = {
+      // clang-format off
+               "OpCapability Shader",
+          "%1 = OpExtInstImport \"GLSL.std.450\"",
+               "OpMemoryModel Logical GLSL450",
+               "OpEntryPoint Fragment %main \"main\" %BaseColor %gl_FragColor",
+               "OpExecutionMode %main OriginUpperLeft",
+               "OpSource GLSL 140",
+               "OpName %main \"main\"",
+               "OpName %foo2_f1_f1_ \"foo2(f1;f1;\"",
+               "OpName %f \"f\"",
+               "OpName %f2 \"f2\"",
+               "OpName %foo_vf4_ \"foo(vf4;\"",
+               "OpName %bar \"bar\"",
+               "OpName %param \"param\"",
+               "OpName %param_0 \"param\"",
+               "OpName %color \"color\"",
+               "OpName %BaseColor \"BaseColor\"",
+               "OpName %param_1 \"param\"",
+               "OpName %gl_FragColor \"gl_FragColor\"",
+       "%void = OpTypeVoid",
+          "%3 = OpTypeFunction %void",
+      "%float = OpTypeFloat 32",
+"%_ptr_Function_float = OpTypePointer Function %float",
+          "%8 = OpTypeFunction %float %_ptr_Function_float %_ptr_Function_float",   
+    "%v4float = OpTypeVector %float 4", 
+"%_ptr_Function_v4float = OpTypePointer Function %v4float",
+         "%15 = OpTypeFunction %float %_ptr_Function_v4float",
+       "%uint = OpTypeInt 32 0", 
+         "%25 = OpConstant %uint 0",
+         "%28 = OpConstant %uint 1",
+         "%34 = OpConstant %uint 2", 
+"%_ptr_Input_v4float = OpTypePointer Input %v4float",
+  "%BaseColor = OpVariable %_ptr_Input_v4float Input",
+"%_ptr_Output_v4float = OpTypePointer Output %v4float",
+"%gl_FragColor = OpVariable %_ptr_Output_v4float Output",
+       "%main = OpFunction %void None %3",
+          "%5 = OpLabel",
+      "%color = OpVariable %_ptr_Function_v4float Function",
+    "%param_1 = OpVariable %_ptr_Function_v4float Function",
+         "%44 = OpLoad %v4float %BaseColor",
+               "OpStore %param_1 %44",
+         "%45 = OpFunctionCall %float %foo_vf4_ %param_1",
+         "%46 = OpCompositeConstruct %v4float %45 %45 %45 %45",
+               "OpStore %color %46",
+         "%49 = OpLoad %v4float %color",
+               "OpStore %gl_FragColor %49",
+               "OpReturn",
+               "OpFunctionEnd",
+"%foo2_f1_f1_ = OpFunction %float None %8",
+          "%f = OpFunctionParameter %_ptr_Function_float",
+         "%f2 = OpFunctionParameter %_ptr_Function_float",
+         "%12 = OpLabel",
+         "%19 = OpLoad %float %f",
+         "%20 = OpLoad %float %f2",
+         "%21 = OpFMul %float %19 %20",
+               "OpReturnValue %21",
+               "OpFunctionEnd",
+   "%foo_vf4_ = OpFunction %float None %15",
```

Comment:
It helps to put the original GLSL source code for this assembly functions in comments.